app/frontend/src/components/Answer/AnswerParser.tsx

```diff
@@ -1,16 +1,44 @@
 import { renderToStaticMarkup } from "react-dom/server";
-import { getCitationFilePath } from "../../api";
+import { ChatAppResponse, getCitationFilePath } from "../../api";
 
 type HtmlParsedAnswer = {
     answerHtml: string;
     citations: string[];
 };
 
-export function parseAnswerToHtml(answer: string, isStreaming: boolean, onCitationClicked: (citationFilePath: string) => void): HtmlParsedAnswer {
+// Function to validate citation format and check if it is a valid citation within the context
+function isCitationValid(contextDataPoints: any, citationCandidate: string): boolean {
+    const regex = /^[^\s]+\.[a-zA-Z0-9]+/;
+    if (!regex.test(citationCandidate)) {
+        return false;
+    }
+
+    // Check if contextDataPoints is an object with a text property that is an array
+    let dataPointsArray: string[];
+    if (Array.isArray(contextDataPoints)) {
+        dataPointsArray = contextDataPoints;
+    } else if (contextDataPoints && Array.isArray(contextDataPoints.text)) {
+        dataPointsArray = contextDataPoints.text;
+    } else {
+        return false;
+    }
+
+    // Check if the citation is included in any of the strings within the text array
+    const isValidCitation = dataPointsArray.some(dataPoint => dataPoint.includes(citationCandidate));
```

Comment:
I think checking startsWith() would also work, right? Given the current response:

```
        "data_points": {
            "text": [
                "Benefit_Options.pdf#page=2:  With Northwind Health Plus, you can choose from a variety of in-network providers, ...
```

Or would startsWith not cover a situation?

Reply:
@pamelafox - Thank you, `startsWith()` worked as well:
![image](https://github.com/user-attachments/assets/d6bdef87-6463-43ab-9ef9-2965f076589c)

(I also noticed another issue: as seen in the screenshot, "Capture the full web page with scroll" in Edge does not seem to work well with the question input field.)

Reply:
Here is the result with` startsWith()`:
<img width="487" alt="image" src="https://github.com/user-attachments/assets/7a9035f9-f494-42ac-899a-88d4ec55b7e3">